mobile/library/common/jni/jni_interface.cc

```diff
@@ -1363,6 +1363,13 @@ extern "C" JNIEXPORT jlong JNICALL Java_io_envoyproxy_envoymobile_engine_JniLibr
     }
     builder.setXds(std::move(xds_builder));
   }
+#else
+  std::string native_xds_address = getCppString(env, xds_address);
+  if (!native_xds_address.empty()) {
+    throwException(env, "java/lang/UnsupportedOperationException",
```

Comment:
nit: It looks like in both the `#ifdef` branches we call getCppString and check if it's non-empty. Consider moving the `#ifdef` inside the  `if (!native_xds_address.empty()) {` branch.